src/backend/cargo.rs

```diff
@@ -23,9 +22,11 @@ impl Backend for Cargo {
             return Ok(BTreeMap::new());
         }
 
-        let file = get_crates_file().context("getting path to crates file")?;
+        let file = dirs::home_dir()
+            .context("getting the home directory")?
```

Comment:
We should first check `$CARGO_HOME` and then use this as the path since this was an issue that was patched upstream. Also, we need to `extend` using `[".cargo", ".crates2.json"]` so the following would give the wrong path ig.